src/utils/AssayGeneratorMetabo.cpp

```diff
@@ -418,7 +413,7 @@ class TOPPAssayGeneratorMetabo :
                                                              sirius_algo);
   
         // sort vector path list
-        std::sort(subdirs.begin(), subdirs.end(), extractAndCompareScanIndexLess_);
+        std::sort(subdirs.begin(), subdirs.end(), SiriusAdapterAlgorithm::extractAndCompareScanIndexLess);
```

Comment:
how many subdirs are we talking about? because for every comparison, the string will be parsed twice...

Reply:
Depending on the files to be analysed 100-100.000. 
Hm, what would be a better solution here?

Reply:
@cbielow  Would it be faster to have a struct e.g. 
```
      struct OPENMS_DLLAPI SiriusWorkspaceIndex
      {
      public:
        String file_path;
        int scan_index;
      };
```
perform the extraction of the scan index once and save them in a `vector<SiriusWorkspaceIndex> siriusworkspaceindex`- and compare the scan indices within the second vector,
```
std::sort(siriusworkspaceindex.begin(), siriusworkspaceindex.end(), 
               [](const auto& i, const auto& j) { return i.scan_index < j.scan_index; } );
```
And separate them later on - or is that too much overhead?

Reply:
yes. to make it a bit more efficient, its enough to store the array_index + scan index in the struct, then sort by scan index and rebuild the subdir vector from the array_indices. But your method also works. Stringviews would also be an option, but probably a bit slower than indices.